code-challenges.test.js

```diff
@@ -20,19 +20,89 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+
+describe("switchedColors", () => {
+    it("takes in an array, removes the first item from the array and shuffles the remaining items", () => {
+        expect(switchedColors(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+        expect(switchedColors(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+})
+
 // b) Create the function that makes the test pass.
+//Created a new variable called switchedColors
+const switchedColors = (array) => {
+//created a new array using the spread operator called swap
+    const [...swap] = array
+//Tried using the fisher-yates method to shuffle the array
+    for (let i = swap.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1))
+        [swap[i], swap[j]] = [swap[j], swap[i]]  
+    }
+//Tried returning swap to get expected outcome but kept getting ReferenceError: Cannot access 'j' before initialization 
+    return swap
+}
```

Comment:
you were really close Jean. I'm not sure why you created a new array.  You just needed to remove the first item.  Here's an article that breaks down Fisher Yates, as well as some other options.  [here](https://www.freecodecamp.org/news/how-to-shuffle-an-array-of-items-using-javascript-or-typescript/) Nice work. I can tell you tried!